tools/c7n_azure/c7n_azure/resources/postgresql_server.py

```diff
@@ -69,3 +71,77 @@ def _query_rules(self, resource):
                 continue
             resource_rules.add(rule)
         return resource_rules
+
+
+@PostgresqlServer.filter_registry.register('configuration-parameters')
+class ConfigurationParametersFilter(ValueFilter):
+    """Filter by configuration parameter for this postresql server
+
+    Configurations are made available to the filter as a map with each
+    key holding the name of the configuration and each value holding
+    the properties of the Configuration as defined here:
+    https://learn.microsoft.com/en-us/python/api/azure-mgmt-rdbms/azure.mgmt.rdbms.postgresql.models.configuration?view=azure-python
+
+    :example:
+
+    Example JSON document showing the data format provided to the filter
+
+    .. code-block:: json
+
+      {
+        log_connections: {
+          "id": "<example-id>",
+          "name": "log_connections",
+          "type": "Microsoft.DBforPostgreSQL/servers/configurations",
+          "value": "off",
+          "description": "Logs each successful connection.",
+          "default_value": "on",
+          "data_type": "Boolean",
+          "allowed_values": "on,off",
+          "source": "user-override"
+        },
+        {
+          "id": "<example-id>",
+          "name": "log_min_duration_statement",
+          "type": "Microsoft.DBforPostgreSQL/servers/configurations",
+          "value": "-1",
+          "description": "example description",
+          "default_value": "-1",
+          "data_type": "Integer",
+          "allowed_values": "-1-2147483647",
+          "source": "system-default"
+        }
+      }
+
+    :example:
+
+    Find Postgresql servers with log_connections not enabled
+
+    .. code-block:: yaml
+
+        policies:
+          - name: sql-database-no-log-connections
+            resource: azure.postgresql-server
+            filters:
+              - type: configuration-parameters
+                key: log_connections.value
+                op: ne
+                value: 'on'
+
+    """
+    schema = type_schema('configuration-parameters', rinherit=ValueFilter.schema)
+
+    def __call__(self, resource):
+        key = 'configurations'
+        if key not in resource['properties']:
+            client = self.manager.get_client()
+            query = client.configurations.list_by_server(
+                resource['resourceGroup'],
+                resource['name']
+            )
+
+            # map the config parameters to an dict to make validating individual values easier
+            configurations = {item.name: vars(item) for item in query or []}
+            resource['properties'][key] = configurations
+
+        return super(ConfigurationParametersFilter, self).__call__(resource['properties'][key])
```

Comment:
python 3 syntax, you don't need to pass the base class for super, also __call__ is typically just done directly with `callable()`

ie, `super()(resource['properties'][key])`

Reply:
Here I just followed the convention I saw being used, but the codebase seems to be pretty split between the two upon doing a search. I'll update it.

Reply:
Also note that `super()` returns a delegate object that isn't directly callable as it doesn't pass through magic methods. It only delegates for explicit `.<name>` syntax - so explicitly calling `__call__` is still required.